maven-cache.sh

```diff
@@ -0,0 +1,16 @@
+#check if maven dependecnies are cached by previous build
+#if its then it downloads the prebuild cache from online source
+# MAVEN_CACHE_TAR_URL : Is for prebuild cache URL. It should be tar.gz
+# MAVEN_CACHE_TAR_FILE : tar file name
+
+#!/bin/sh
+
+if [ ! -d /root/.mvnrepository/repository ]; then
+    echo "downloaidng the prebuilt maven cache\n"
+
+    curl -LO ${MAVEN_CACHE_TAR_URL} \
+        && tar -xvzf ${MAVEN_CACHE_TAR_FILE}.tar.gz \
+        && mv $PWD/${MAVEN_CACHE_TAR_FILE}/repository /root/.mvnrepository \
+        && rm -rf $PWD/${MAVEN_CACHE_TAR_FILE}.tar.gz
+fi
+sleep infinity
```

Comment:
nit: you may want to `exec sleep infinity here` so the shell script process get replaced